src_kt/javatests/arcs/util/GuardTest.kt

```diff
@@ -0,0 +1,55 @@
+package arcs.util
+
+import arcs.arcs.util.guardWith
+import arcs.testutil.assertThrows
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+/** Tests for the [guardWith]-generated property delegate: [GuardDelegate]. */
+@ExperimentalCoroutinesApi
+@RunWith(JUnit4::class)
+class GuardTest {
+    class RequiresLocking(initialValue: Int = 0) {
+        val mutex = Mutex()
+        var value by guardWith(mutex) { initialValue }
+    }
+
+    @Test
+    fun accessingGuardedValue_outsideOfLock_throws() {
+        val obj = RequiresLocking()
+
```

Comment:
I like having this runtime check, but I still feel like someone could check in code like this that lays dormant and unexercised for a long time.  I have no objection to it, but I feel like it's complimentary to @GuardedBy? Am I the only one?

Reply:
Definitely complimentary, I like that it'll cause unit tests to fail if someone tries to do something with the value without locking.  We can look into a custom lint rule to replicate the behavior of `@GuardedBy`.  Or we could just use `@GuardedBy` as well, though that's a bit ugly.